script/tool/lib/src/gradle_check_command.dart

```diff
@@ -44,16 +44,50 @@ class GradleCheckCommand extends PackageLoopingCommand {
 
   bool _validateBuildGradle(RepositoryPackage package,
       {required bool isExample}) {
-    // Currently the only check is not relevant to examples; checks that apply
-    // to both plugins and examples should go above here.
-    if (!isExample) {
-      print('${indentation}Validating android/build.gradle.');
-      final String contents = package
-          .platformDirectory(FlutterPlatform.android)
-          .childFile('build.gradle')
-          .readAsStringSync();
-      final List<String> lines = contents.split('\n');
+    final Directory androidDir =
+        package.platformDirectory(FlutterPlatform.android);
+    // For apps, the relevant files are in android/app, rather than android/.
+    final Directory parentDir =
+        isExample ? androidDir.childDirectory('app') : androidDir;
+    final File gradleFile = parentDir.childFile('build.gradle');
```

Comment:
Nonblocking: Can you extract this into a method that gives you the gradle file? `parentDir` makes sense as a concept when getting a build.gradle file but I think will make this code harder to reason about when we end up doing validations on multiple build.gradle files in a single example app. 

It is way more refactoring than I would want to ask from you but I think the right structure of this validation in the long term will be a top line _validateBuildGradle method that passe the contents of build.gradle files into 3 different validators depending on what the project has. One for plugins, one for the root of an application and one for android/app/build.gradle. Then if we need to validate something specific like namespace there is a method for that called in every sub validator.

Reply:
> Nonblocking: Can you extract this into a method that gives you the gradle file?

Good idea, done.

> It is way more refactoring than I would want to ask from you but I think the right structure of this validation [...]

https://xkcd.com/356/ 😁